### PR TITLE
Fix auto assign played after rules in tickets

### PR DESCRIPTION
**Pull request by RomainLvr**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !32840

- I create a business rule for tickets when they are added that, as soon as a technician group exists in the ticket, the ticket is assigned the status new.
- I assign a group responsible for the category.
- I set 'Automatic assignment of tickets, changes and problems' to "category then material" from my entity's support parameters. 

When the ticket is created, the group assignments are correct, but the rule assigning a status to the ticket doesn't work.
This is due to the fact that the rule is launched before the automatic assignment in the ticket addition procedure.

### CODE REVIEW

**Comment by trasher** [comment]
Existing tests are broken.

Also, I'm not sure this kind of issue really can be handled right now; it should cause other issues to change the order of execution :(

File: src/Rule.php [hunk 1]
```diff
@@ -1538,6 +1538,15 @@ public function process(&$input, &$output, &$params, &$options = [])
                 $hook_params["output"]   = $output;
                 Plugin::doHook(Hooks::RULE_MATCHED, $hook_params);
                 $output["_rule_process"] = true;
+
+                //If the rule has changed the ticket status, add a param to avoid another status change
+                if ($this instanceof RuleTicket) {
+                    foreach ($this->actions as $action) {
+                        if ($action->fields['field'] === 'status') {
+                            $output['_status_changed'] = true;
```
Comment (Rom1-B) [inline src/Rule.php:1546]: `isStatusComputationBlocked()` will check `_do_not_compute_status`

```suggestion
                            $output['_do_not_compute_status'] = true;
```

**Review by Rom1-B — CHANGES_REQUESTED** [review]

File: src/Rule.php (hunk 1 above)
Comment (RomainLvr) [reply src/Rule.php]: Thank you! applied :+1: [9573779](https://github.com/glpi-project/glpi/pull/17088/commits/95737792deeb55180829debceba54ebb198d8223)

**Review by cedric-anne — COMMENTED** [review]
There is already a piece of code that is supposed to handle this case: https://github.com/glpi-project/glpi/blob/fae0c077b9c0ed9111d6e6666102f63e97be5f3b/src/RuleTicket.php#L206-L210

Could you check why it does not work ?

**Comment by RomainLvr** [comment]
> There is already a piece of code that is supposed to handle this case:
> 
> https://github.com/glpi-project/glpi/blob/fae0c077b9c0ed9111d6e6666102f63e97be5f3b/src/RuleTicket.php#L206-L210
> 
> Could you check why it does not work ?

Indeed this case was already treated and it works, the problem still lies in what is mentioned in this ticket: !32835

**Comment by RomainLvr** [comment]
> > There is already a piece of code that is supposed to handle this case:
> > https://github.com/glpi-project/glpi/blob/fae0c077b9c0ed9111d6e6666102f63e97be5f3b/src/RuleTicket.php#L206-L210
> > 
> > Could you check why it does not work ?
> 
> Indeed this case was already treated and it works, the problem still lies in what is mentioned in this ticket: !32835

Finally, the first concern is that the rule is not executed at all, because one of the criteria required for this rule is not registered until after the rule has been attempted to be executed. As a result, the '__do_not_compute_status_' is never set, and the status lock doesn't work.

**Comment by cedric-anne** [comment]
Same comment as in #16885. We have to deal with current rules engine limitations.

**Pull request closed without merging**